barretenberg/cpp/pil/avm/avm_mini.pil

```diff
@@ -131,26 +130,36 @@ namespace avmMini(256);
     //===== CALL_RETURN ========================================================
     // The program counter in the next row should be equal to the value loaded from the ia register
     // This implies that a load from memory must occur at the same time
-    // Imply that we must load the jump selector into mema
-    #[return_pointer_increment]
+    // Imply that we must load the return location into mem_idx_a
+    
+    #[RETURN_POINTER_INCREMENT]
     sel_internal_call * ( internal_return_ptr' - ( internal_return_ptr + 1)) = 0; // we increment our jump selector
     sel_internal_call * ( internal_return_ptr - mem_idx_a) = 0;
-    sel_internal_call * (rwa - 1) = 0; // it must be a write
-    sel_internal_call * (mem_op_a - 1) = 0;
     sel_internal_call * ((pc + 1) - ia) = 0;
+    // TODO(md): Below relations may be removed through sub-op table lookup
+    sel_internal_call * (rwa - 1) = 0;
+    sel_internal_call * (mem_op_a - 1) = 0;
     
     // We must load the memory pointer to be the  internal_return_ptr 
-    #[return_pointer_decrement]
+    #[RETURN_POINTER_DECREMENT]
     sel_internal_return * ( internal_return_ptr' - ( internal_return_ptr - 1)) = 0; // We decrement out jump selector
     sel_internal_return * ( (internal_return_ptr - 1) - mem_idx_a) = 0;
     sel_internal_return * (pc' - ia) = 0;
+
+    // TODO(md): Below relations may be removed through sub-op table lookup
     sel_internal_return * rwa = 0;
     sel_internal_return * (mem_op_a - 1) = 0;
 
+    //===== CONTROL_FLOW_CONSITENCY ============================================
+    pol CONTROL_FLOW_SELECTORS = (first + sel_internal_call + sel_internal_return + sel_halt);
+    pol OPCODE_SELECTORS = (sel_op_add + sel_op_sub + sel_op_div + sel_op_mul);
+
     // Program counter must increment if not jumping or returning
-    #[pc_increment]
-    (1-first) * (1- sel_halt) * (sel_op_add + sel_op_sub + sel_op_div + sel_op_mul) * (pc' - (pc + 1)) = 0;
+    #[PC_INCREMENT]
+    (1 - first) * (1 - sel_halt) * OPCODE_SELECTORS * (pc' - (pc + 1)) = 0;
 
+    #[INTERNAL_RETURN_POINTER_CONSISTENCY]
+    (1 - CONTROL_FLOW_SELECTORS) * (internal_return_ptr' - internal_return_ptr) = 0; 
```

Comment:
I suggest a comment line on top of the relation with this:
// first == 0 && sel_internal_call == 0  &&  sel_internal_return == 0 && sel_halt == 0 ==> internal_return_ptr == internal_return_ptr'